06 - Type Ahead/index-START.html

```diff
@@ -15,8 +15,68 @@
     </ul>
   </form>
 <script>
-const endpoint = 'https://gist.githubusercontent.com/Miserlou/c5cd8364bf9b2420bb29/raw/2bf258763cdddd704f8ffd3ea9a3e81d25e2c6f6/cities.json';
+  //// global variables
+  const endpoint = 'https://gist.githubusercontent.com/Miserlou/c5cd8364bf9b2420bb29/raw/2bf258763cdddd704f8ffd3ea9a3e81d25e2c6f6/cities.json';
+  let cities = [];
+  let searchTerm = '';
 
+  //// selectors
+  const form = document.querySelector('.search-form');
+  const input = document.querySelector('.search');
+  const suggestions = document.querySelector('.suggestions');
+
+  //// callbacks
+  // fetchData() is async if we decide to also fetch on change or on submit for example
+  const fetchData = async (URL) => {
+    try {
+      const res = await fetch(URL);
+      const data = await res.json();
+      cities = [...data];
+    } catch (error) {
+      const isError = () => {
+        const message = `<h2 class="errorMessage">${error}</h2>`;
+        suggestions.innerHTML = message;  
+      }
+      isError();
+    }
+  }
+
+  fetchData(endpoint);
+
+  const findMatch = (wordToSearch, dataToFilter) => {
```

Comment:
let's use plain functions (as discussed in #5) here. or we go a step further and use classes

Reply:
ok changed